libs/libapi/src/libapi/duckdb.py

```diff
@@ -60,13 +64,27 @@ async def get_index_file_location_and_download_if_missing(
         return index_file_location
 
 
-def get_download_folder(root_directory: StrPath, dataset: str, revision: str, config: str, split: str) -> str:
+def get_download_folder(
+    root_directory: StrPath, size_bytes: int, dataset: str, revision: str, config: str, split: str
+) -> str:
+    if not available_disk_space(root_directory, size_bytes):
+        raise DownloadIndexError(
+            "Cannot perform the search due to a lack of disk space on the server. Please report the issue."
+        )
     payload = (dataset, config, split, revision)
     hash_suffix = sha1(json.dumps(payload, sort_keys=True).encode(), usedforsecurity=False).hexdigest()[:8]
     subdirectory = "".join([c if re.match(r"[\w-]", c) else "-" for c in f"{dataset}-{hash_suffix}"])
     return f"{root_directory}/{DUCKDB_INDEX_DOWNLOADS_SUBDIRECTORY}/{DUCKDB_VERSION}/{subdirectory}"
 
 
+def available_disk_space(path: StrPath, required_space: int) -> bool:
+    disk_stat = os.statvfs(path)
+    # Calculate free space in bytes
+    free_space = disk_stat.f_bavail * disk_stat.f_frsize
+    logging.debug(f"{free_space} available space, needed {required_space}")
+    return free_space >= required_space
```

Comment:
nit (feel free to ignore): we could also rename to `check_available_disk_space` and raise DownloadIndexError here, instead of returning a boolean. It would simplify the code a bit

Reply:
Done. Indeed, it is simpler, thanks!